client/my-sites/checkout/composite-checkout/components/checkout-main.tsx

```diff
@@ -114,8 +116,10 @@ export default function CheckoutMain( {
 	// `getThankYouUrl`.
 	onAfterPaymentComplete?: () => void;
 	disabledThankYouPage?: boolean;
+	isAkismetSitelessCheckout?: boolean;
 	isJetpackCheckout?: boolean;
 	jetpackSiteSlug?: string;
+	akismetSiteSlug?: string;
```

Comment:
From a Payments perspective, I think that would be great! Teams have been adding these booleans for a long time and I'd love to have fewer of them.

I would suggest using a simple string union for the options (like we did with [contact details type](https://github.com/Automattic/wp-calypso/blob/136443bd1dfb5792563475a4a4981e0ffded1750/packages/wpcom-checkout/src/types.ts#L245)) instead of a TS enum because the latter is a bit of a hack (TS normally can be erased but enums are actually code and this may create problems if JS implements enums in the future).